superduperdb/base/datalayer.py

```diff
@@ -373,6 +347,20 @@ def predict(
             return Document(out), [Document(x) for x in context]
         return Document(out), []
 
+    def validate(self, query: ExecuteQuery, *args, **kwargs) -> None:
+        """
+        Validate a query on the db.
+
+        :param query: select, insert, delete, update,
+        """
+        if not isinstance(query, (Delete, Select, Update)):
+            return
+
+        if query.variables:
+            query.set_variables(self).validate(self)
+        else:
+            query.validate(self)
```

Comment:
This still can be a shortened a bit. What do you think?
```suggestion
        if query.variables:
            query = query.set_variables(self)

        query.validate(self)
```

Reply:
yes